runtime/src/main/java/org/corfudb/runtime/view/workflows/WorkflowRequest.java

```diff
@@ -141,6 +141,9 @@ public void invoke() {
                     throw new IllegalStateException("Orchestrator can not be selected");
                 }
 
+            } catch(RuntimeException e) {
```

Comment:
This catch will mask the next catch block, because NetworkException and IllegalStateException are also RuntimeException, you need to put it after the next catch block. Please also check with @PavelZaytsev if initially he intentionally wanted to not retry on other runtime exceptions, as there might be some edge cases.

Reply:
Moved the catch block. It was gonna mask other exceptions.